### PR TITLE
Resource cloning removal

### PR DESCRIPTION
**Pull request by juanconde**
Chef resources invocation (from providers) rewritten to avoid cloning (forbidden in chef-client 13.x)

### CODE REVIEW

File: providers/email_setup.rb [hunk 1]
```diff
@@ -69,12 +69,22 @@ def setup_gecos_as_default_profile(profile_ini)
     if os_supported? &&
        (policy_active?('users_mgmt', 'email_setup_res') ||
         policy_autoreversible?('users_mgmt', 'email_setup_res'))
+ 
+      directory '/var/cache/gecos' do
```
Comment (alfem) [inline providers/email_setup.rb]: Probably unneeded if we use 'recursive' parameter in deeper directory

See https://docs.chef.io/resource_directory.html

File: providers/email_setup.rb [hunk 2]
```diff
@@ -250,7 +252,8 @@ def setup_gecos_as_default_profile(profile_ini)
           plugin_file = "/var/cache/gecos/email_setup/#{plugin_file_name}"
 
           # Download extension if necessary
-          remote_file plugin_file do
+          remote_file '#{plugin_file}-#{username}' do
```
Comment (alfem) [inline providers/email_setup.rb]: I think extensions must be identified as unique resources using "plugin-file", not "plugin-file" and "username"

File: providers/email_setup.rb (hunk 1 above)
Comment (juanconde) [reply providers/email_setup.rb]: I agree. I simply moved old code out of a loop. I'll fix it.

File: providers/email_setup.rb (hunk 2 above)
Comment (juanconde) [reply providers/email_setup.rb]: Presently plugins are installed inside a users loop (starting in line 85), thus "username" is required in order to have unique resource names. I'll check if plugins download can be moved outside the users loop.

Comment (juanconde) [reply providers/email_setup.rb]: Plugins to be installed are included in email templates, there can be n different templates in the databag, the template to be used is especified per user. Conclusion: plugin download can not be taken out of the users loop.

File: providers/email_setup.rb (hunk 1 above)
Comment (juanconde) [reply providers/email_setup.rb]: Fixed in PR #257

File: providers/email_setup.rb (hunk 2 above)
Comment (alfem) [reply providers/email_setup.rb]: Ok, let's merge and test all these changes

**Pull request merged**